### PR TITLE
test: add local npm registry to failing tests

### PR DESCRIPTION
**Pull request by alan-agius4**

### CODE REVIEW

File: tests/legacy-cli/e2e/tests/update/update-1.7.ts [hunk 1]
```diff
@@ -7,11 +7,11 @@ import { expectToFail } from '../../utils/utils';
 export default async function() {
   const extraUpdateArgs = (await isPrereleaseCli()) ? ['--next', '--force'] : [];
 
-  await createProjectFromAsset('1.7-project');
+  await createProjectFromAsset('1.7-project', true);
   fs.writeFileSync('.npmrc', 'registry = http://localhost:4873', 'utf8');
 
   await useCIChrome('.');
-  await expectToFail(() => ng('build'));
+  await ng('build');
```
Comment (alan-agius4) [inline tests/legacy-cli/e2e/tests/update/update-1.7.ts]: Not sure why this was expected to fail. It should succeed!

File: tests/legacy-cli/e2e/tests/update/update-1.7-longhand.ts [hunk 2]
```diff
@@ -7,10 +7,10 @@ import { expectToFail } from '../../utils/utils';
 export default async function() {
   const extraUpdateArgs = (await isPrereleaseCli()) ? ['--next', '--force'] : [];
 
-  await createProjectFromAsset('1.7-project');
+  await createProjectFromAsset('1.7-project', true);
   fs.writeFileSync('.npmrc', 'registry = http://localhost:4873', 'utf8');
 
-  await expectToFail(() => ng('build'));
+  await ng('build');
```
Comment (alan-agius4) [inline tests/legacy-cli/e2e/tests/update/update-1.7-longhand.ts]: Not sure why this was expected to fail. It should succeed!

File: tests/legacy-cli/e2e/tests/update/update-1.7.ts (hunk 1 above)
Comment (alan-agius4) [inline tests/legacy-cli/e2e/tests/update/update-1.7.ts]: Same

File: tests/legacy-cli/e2e/tests/misc/ask-analytics.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ export default async function() {
     // Install the CLI with TTY force enabled
     const execution = execWithEnv(
       'npm',
-      ['install', packages['@angular/cli'].tar],
+      ['install', packages['@angular/cli'].tar, '--registry=http://localhost:4873'],
```
Comment (clydin) [inline tests/legacy-cli/e2e/tests/misc/ask-analytics.ts]: You can use the package name directly

File: tests/legacy-cli/e2e/tests/schematics_cli/basic.ts [hunk 4]
```diff
@@ -12,7 +12,13 @@ export default async function () {
   }
 
   const startCwd = process.cwd();
-  await silentNpm('install', '-g', packages['@angular-devkit/schematics-cli'].tar, '--unsafe-perm');
+  await silentNpm(
+    'install',
+    '-g',
+    packages['@angular-devkit/schematics-cli'].tar,
+    '--unsafe-perm',
```
Comment (clydin) [inline tests/legacy-cli/e2e/tests/schematics_cli/basic.ts]: Same here for the package name.
Also, is unsafe-perm actually needed?

File: tests/legacy-cli/e2e/utils/assets.ts [hunk 5]
```diff
@@ -39,10 +39,12 @@ export function copyAssets(assetName: string) {
...
-  return Promise.resolve()
-    .then(() => copyAssets(assetName))
-    .then(dir => process.chdir(dir))
-    .then(() => useNpmPackages ? null : useBuiltPackages())
-    .then(() => silentNpm('install'));
+export async function createProjectFromAsset(assetName: string, useNpmPackages = false) {
+  const dir = await copyAssets(assetName);
+  process.chdir(dir);
+  if (!useNpmPackages) {
+    await useBuiltPackages();
```
Comment (clydin) [inline tests/legacy-cli/e2e/utils/assets.ts]: Technically, no test should be using the built packages directly anymore but this fixes the problem for now.

We should revisit these tests though.  Potentially using `npx` to generate a new project of the appropriate version directly.

**Pull request merged**